FlingEngine/Gameplay/src/FirstPersonCamera.cpp

```diff
@@ -0,0 +1,83 @@
+#include "FirstPersonCamera.h"
+#include "FlingMath.h"
+#include "pch.h"
+
+namespace Fling 
+{
+    FirstPersonCamera::FirstPersonCamera(float aspectRatio) :
+		m_front(glm::vec3(-1.0f))
+    {
+		m_nearPlane   =   0.1f;
+		m_farPlane    =   1000.0f;
+		m_speed       =   100.0f;
+        m_aspectRatio =   aspectRatio;
+		m_fieldOfView =   glm::radians(45.0f);
+		m_position    =   glm::vec3(2.0f, 2.0f, 2.0f);
+		m_rotation    =   glm::vec3(0.0f, 0.0f, 0.0f);
+		m_worldUp     =   glm::vec3(0.0f, 0.0f, 1.0f);
+    }
+
+
+	void FirstPersonCamera::Update(float dt)
+    {
+        float moveSpeed = m_speed * dt;
+
+        if(Input::IsKeyHeld(KeyNames::FL_KEY_W))
+			m_position += m_front * moveSpeed;
+
+        if(Input::IsKeyHeld(KeyNames::FL_KEY_S))
+			m_position -= m_front * moveSpeed;
+
+        if(Input::IsKeyHeld(KeyNames::FL_KEY_A))
+			m_position -= m_right* moveSpeed;
+
+        if(Input::IsKeyHeld(KeyNames::FL_KEY_D))
+			m_position += m_right * moveSpeed;
+
+		//Rotation
```

Comment:
another different indent? Looks like somehow we switched from tabs to spaces or something like that. Doesn't really bother me, just a weird thing you might wanna check out in your dev evniornment